cmd/neofs-node/validate.go

```diff
@@ -1,71 +1,103 @@
 package main
 
 import (
+	"errors"
 	"fmt"
 	"path/filepath"
 
 	"github.com/nspcc-dev/neofs-node/cmd/neofs-node/config"
 	engineconfig "github.com/nspcc-dev/neofs-node/cmd/neofs-node/config/engine"
 	shardconfig "github.com/nspcc-dev/neofs-node/cmd/neofs-node/config/engine/shard"
-	loggerconfig "github.com/nspcc-dev/neofs-node/cmd/neofs-node/config/logger"
 	"github.com/nspcc-dev/neofs-node/pkg/local_object_storage/blobstor/fstree"
 	"github.com/nspcc-dev/neofs-node/pkg/local_object_storage/blobstor/peapod"
 	"go.uber.org/zap/zapcore"
 )
 
+var (
+	errEndpointNotSet = errors.New("empty/not set endpoint, see `grpc.endpoint` section")
+	errTLSKeyNotSet   = errors.New("empty/not set TLS key file path, see `grpc.tls.key` section")
+	errTLSCertNotSet  = errors.New("empty/not set TLS certificate file path, see `grpc.tls.certificate` section")
+)
+
 // validateConfig validates storage node configuration.
 func validateConfig(c *config.Config) error {
 	// logger configuration validation
 
-	_, err := zapcore.ParseLevel(loggerconfig.Level(c))
+	_, err := zapcore.ParseLevel(c.Logger.Level)
 	if err != nil {
 		return fmt.Errorf("invalid logger level: %w", err)
 	}
 
-	logEncoding := loggerconfig.Encoding(c)
+	logEncoding := c.Logger.Encoding
 	if logEncoding != "console" && logEncoding != "json" {
 		return fmt.Errorf("invalid logger encoding: %s", logEncoding)
 	}
 
+	// chain configuration validation
+
+	if len(c.FSChain.Endpoints) == 0 {
+		return errors.New("no FS chain RPC endpoints, see `fschain.endpoints` section")
+	}
+
+	// grpc configuration validation
+
+	if len(c.GRPC) == 0 {
+		return errors.New("no gRPC endpoints, see `grpc` section")
+	}
+	for i := range c.GRPC {
+		if c.GRPC[i].Endpoint == "" {
+			return errEndpointNotSet
+		}
+		if c.GRPC[i].TLS.Enabled {
+			if c.GRPC[i].TLS.Certificate == "" {
+				return errTLSCertNotSet
+			}
+			if c.GRPC[i].TLS.Key == "" {
+				return errTLSKeyNotSet
+			}
+		}
+	}
+
 	// shard configuration validation
 
 	shardNum := 0
 	paths := make(map[string]pathDescription)
-	return engineconfig.IterateShards(c, false, func(sc *shardconfig.Config) error {
-		if sc.WriteCache().Enabled() {
-			err := addPath(paths, "writecache", shardNum, sc.WriteCache().Path())
+	return engineconfig.IterateShards(&c.Storage, false, func(sc *shardconfig.ShardDetails) error {
+		if sc.Mode.String() == "UNDEFINED" {
```

Comment:
I'd expect some non-string comparison here. Maybe even `!sc.Mode.IsValid()` that would do the magic internally.